packages/wrangler/src/traverse-module-graph.ts

```diff
@@ -0,0 +1,112 @@
+import { readFile } from "node:fs/promises";
+import path from "node:path";
+import chalk from "chalk";
+import * as esbuild from "esbuild";
+import { CommonESBuildOptions } from "./bundle";
+import { logger } from "./logger";
+import createModuleCollector from "./module-collection";
+import type { BundleResult } from "./bundle";
+import type { Config } from "./config";
+import type { Entry } from "./entry";
+
+export default async function traverseModuleGraph(
+	entry: Entry,
+	rules: Config["rules"],
+	tsconfig?: string | undefined
+): Promise<BundleResult> {
+	const entrypoint = path.relative(entry.directory, entry.file);
+	// The fact that we're using esbuild to try and traverse the dependency graph to find modules
+	// to upload is an internal detail & optimisation. Errors from esbuild should be ignored, and the worker
+	// uploaded without change.
+	try {
+		const moduleCollector = createModuleCollector({
+			// These are deprecated, and so won't be supported for `--no-bundle` module collection
+			wrangler1xlegacyModuleReferences: {
+				rootDirectory: entry.directory,
+				fileNames: new Set(),
+			},
+			format: entry.format,
+			rules,
+			preserveFileNames: true,
+		});
+		const result = await esbuild.build({
+			entryPoints: [entry.file],
+			absWorkingDir: entry.directory,
+			metafile: true,
+			// This is required for ESBuild to collect modules
+			bundle: true,
+			target: CommonESBuildOptions.target,
+			// Ignore external packages
+			packages: "external",
+			write: false,
+			loader: CommonESBuildOptions.loader,
+			...(tsconfig && { tsconfig }),
+			plugins: [moduleCollector.plugin],
+		});
+
+		const modules = Object.entries((result.metafile as esbuild.Metafile).inputs)
```

Comment:
You use `Object.entries()` here but then always ignore the value of the entry.
Why not just use `Object.keys()`?